compiler/src/dotty/tools/dotc/classpath/ClassPathFactory.scala

```diff
@@ -52,14 +52,23 @@ class ClassPathFactory {
 
   // Internal
   protected def classesInPathImpl(path: String, expand: Boolean)(using Context): List[ClassPath] =
-    for {
+    val files = for {
       file <- expandPath(path, expand)
       dir <- {
         def asImage = if (file.endsWith(".jimage")) Some(AbstractFile.getFile(file)) else None
         Option(AbstractFile.getDirectory(file)).orElse(asImage)
       }
     }
-    yield newClassPath(dir)
+    yield dir
+
+    val expanded = for file <- files
+        a <- ClassPath.expandManifestPath(file.absolutePath)
+      yield
+        newClassPath(AbstractFile.getFile(a.getPath().toString()))
```

Comment:
it seems some of these are null

Reply:
so the issue was that Class-Path manifest entry isn't always correct - so now ignore non-existent entries:
case in point - `org.scala-lang:scala-compiler:2.13.14` lists `scala-reflect.jar` - but this is only correct when specifically bundled in the Scala binary distribution - and not when it comes from a maven cache.

Also only enable the behaviour under the `scala.expandjavacp=true` property, which is enabled in scripts, and should be backwards compatible